tools/DevDiagnostics/DevHome.DevDiagnostics/Helpers/LoaderSnapAssistantTool.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using CommunityToolkit.Mvvm.Input;
+using DevHome.Common.Extensions;
+using DevHome.DevDiagnostics.Controls;
+using DevHome.DevDiagnostics.Models;
+using DevHome.DevDiagnostics.Services;
+using DevHome.Service;
+using Microsoft.Diagnostics.Tracing;
+using Microsoft.Diagnostics.Tracing.Session;
+using Microsoft.UI.Xaml;
+using Microsoft.Win32;
+
+namespace DevHome.DevDiagnostics.Helpers;
+
+public class LoaderSnapAssistantTool
+{
+    private const string WindowsImageETWProvider = "2cb15d1d-5fc1-11d2-abe1-00a0c911f518"; /*EP_Microsoft-Windows-ImageLoad*/
+    private const uint LoaderSnapsFlag = 0x80; /* ETW_UMGL_LDR_SNAPS_FLAG */
+    private readonly DDInsightsService _insightsService;
+    private readonly IDevHomeService _devHomeService;
+    private readonly Microsoft.UI.Dispatching.DispatcherQueue _dispatcher;
+    private readonly Dictionary<int, string> _loaderSnapFailures = new();
+
+    public LoaderSnapAssistantTool()
+    {
+        _dispatcher = Microsoft.UI.Dispatching.DispatcherQueue.GetForCurrentThread();
+        _insightsService = Application.Current.GetService<DDInsightsService>();
+        _devHomeService = Application.Current.GetService<IDevHomeService>();
+        Init();
+    }
+
+    private void Init()
+    {
+        var crashDumpAnalyzerThread = new Thread(() =>
+        {
+            MyETWListener();
+        });
```

Comment:
Pretty sure this can be simplified even further to one line... something like this...

var crashDumpAnalyzerThread = new Thread(MyETWListener);

Reply:
You're right... I'm getting lamda happy. :)